packages/esm-appointments-app/src/form/appointments-form.component.tsx

```diff
@@ -241,10 +241,23 @@ const AppointmentsForm: React.FC<AppointmentsFormProps> = ({
   })();
 
   // Same for creating and editing
-  const handleSaveAppointment = (data: AppointmentFormData) => {
+  const handleSaveAppointment = async (data: AppointmentFormData) => {
     setIsSubmitting(true);
     // Construct appointment payload
     const appointmentPayload = constructAppointmentPayload(data);
+
+    // check if Duplicate Response Occurs
+    const response: FetchResponse = await checkAppointmentConflict(appointmentPayload);
+    if (response.status === 200) {
+      setIsSubmitting(false);
+      showSnackbar({
+        isLowContrast: true,
+        kind: 'error',
+        title: t('appointmentDuplicateForm', 'Appointment is not available'),
```

Comment:
I haven't tested, but looking at the backend code, it looks like this will return a response object with the type of conflict that has been found.  Can we parse the response object and display a more helpful and more accurate error message?  I see two conflict types:  SERVICE_UNAVAILABLE, PATIENT_DOUBLE_BOOKING.  Could we display a message like "Service unavailable at this time" for the first tyoe of conlict, and "Patient already booked for an appointment at this time" in the second of the second